internal/services/iothub/iothub_endpoint_storage_container_resource.go

```diff
@@ -95,8 +97,11 @@ func resourceIotHubEndpointStorageContainer() *pluginsdk.Resource {
 			},
 
 			"encoding": {
-				Type:     pluginsdk.TypeString,
-				Optional: true,
+				Type:             pluginsdk.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          string(devices.EncodingAvro),
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
instead of suppressing case, could we correct the casing on read and store the correct version?